src/design_docs.md

```diff
@@ -0,0 +1,19 @@
+# 🔬 Design documents
+
+The **design documents** (or "design docs", more commonly) describe potential designs. These docs vary greatly in terms of their readiness to be implemented:
+
+* Early on, they describe a vague idea for a future. Often this takes the shape of capturing constraints on the solution, rather than the solution itself.
+* When a feature is getting ready to ship, they can evolve into a full blown RFC, with links to tracking issues or other notes.
```

Comment:
Could have links to RFC procedures: https://github.com/rust-lang/rfcs?tab=readme-ov-file#what-the-process-is

Is it worth mentioning MCPs, etc?

Reply:
Yeah I think it make sense thanks for the suggestion